capstone-project/src/main/java/com/google/sps/data/Marker.java

```diff
@@ -125,9 +125,9 @@ public static Marker fromEntity(Entity entity){
                 .build();
     }
 
-    /** Creates a marker entity from a Marker */
-    public static Entity toEntity(Marker marker){
-        Entity markerEntity = new Entity("Marker");
+    // Copies the fields of a Marker class instance to a Marker entity.
+    public static Entity toEntity(Marker marker, Entity markerEntity){
+
```

Comment:
I liked it better when the creation of the class was here. 
Why did you change it?

Reply:
I wanted to distinguish between two use cases of this method - 1. Storing a new marker which means creating a new entity 2. Updating an existing entity. When the creation of the class was here - this function created a new entity each time it was called. Now I can create a new entity in the storeMarker method and retrieve one in the updateMarker method (as explained in the comment below), and then pass it to set the properties here, in the toEntity method. 
Is there a better alternative?

Reply:
You can overload methods, having:

public static Entity toEntity(Marker marker);
public static Entity toEntity(Marker marker, Entity entity);

I suggest to better document that too.